packages/react-router/lib/router/router.ts

```diff
@@ -5179,10 +5259,14 @@ async function callDataStrategyImpl(
 ): Promise<Record<string, DataStrategyResult>> {
   // Ensure all lazy/lazyMiddleware async functions are kicked off in parallel
   // before we await them where needed below
-  let loadMiddlewarePromise = loadLazyMiddlewareForMatches(matches, manifest);
-  let loadRouteDefinitionsPromises = matches.map((m) =>
+  let loadMiddlewarePromise = loadLazyMiddlewareForMatches(
+    matches,
+    manifest,
+    mapRouteProperties
+  );
+  let loadLazyRoutePromises = matches.map((m) =>
     m.route.lazy
-      ? loadLazyRouteModule(m.route, mapRouteProperties, manifest)
+      ? loadLazyRoute(m.route, manifest, mapRouteProperties)
```

Comment:
This will run all `lazy.*` functions here right?  I think that's the right approach to start - but curious if you were thinking we should take it farther/more granular and do the type of stuff we do in framework mode for library mode where we skip the hydrate fallback if we're not hydrating?

Reply:
Yep, I wanted to do that as a follow-up.